src/main/java/com/celements/web/plugin/cmd/ExternalJavaScriptFilesCommand.java

```diff
@@ -35,127 +40,226 @@
+  @NotNull
   public String addExtJSfileOnce(String jsFile) {
-    return addExtJSfileOnce(jsFile, null);
+    return addExtJSfileOnce(new ExtJsFileParameter()
+        .setJsFile(jsFile));
   }
 
+  /**
+   * @deprecated since 5.4 instead use {@link addExtJSfileOnce(ExtJsFileParameter)}
+   */
+  @Deprecated
+  @NotNull
   public String addExtJSfileOnce(String jsFile, String action) {
-    return addExtJSfileOnce(jsFile, action, null);
+    return addExtJSfileOnce(new ExtJsFileParameter()
+        .setJsFile(jsFile)
+        .setAction(action));
   }
 
+  /**
+   * @deprecated since 5.4 instead use {@link addExtJSfileOnce(ExtJsFileParameter)}
+   */
+  @Deprecated
+  @NotNull
   public String addExtJSfileOnce(String jsFile, String action, String params) {
-    if (!extJSAttUrlSet.contains(jsFile)) {
-      if (getAttUrlCmd().isAttachmentLink(jsFile) || getAttUrlCmd().isOnDiskLink(jsFile)) {
-        extJSAttUrlSet.add(jsFile);
+    return addExtJSfileOnce(new ExtJsFileParameter()
+        .setJsFile(jsFile)
+        .setAction(action)
+        .setParams(params));
+  }
+
+  @NotNull
+  public String addExtJSfileOnce(ExtJsFileParameter extJsFileParams) {
+    if (!extJSAttUrlSet.contains(extJsFileParams.getJsFile())) {
+      if (getAttUrlCmd(extJsFileParams.getAttUrlCmd()).isAttachmentLink(extJsFileParams.getJsFile())
+          || getAttUrlCmd(extJsFileParams.getAttUrlCmd())
+              .isOnDiskLink(extJsFileParams.getJsFile())) {
+        extJSAttUrlSet.add(extJsFileParams.getJsFile());
       }
       String attUrl;
-      if (!StringUtils.isEmpty(action)) {
-        attUrl = getAttUrlCmd().getAttachmentURL(jsFile, action, context);
+      if (!StringUtils.isEmpty(extJsFileParams.getAction())) {
+        attUrl = getAttUrlCmd(extJsFileParams.getAttUrlCmd()).getAttachmentURL(
+            extJsFileParams.getJsFile(),
+            extJsFileParams.getAction(),
+            getModelContext().getXWikiContext());
       } else {
-        attUrl = getAttUrlCmd().getAttachmentURL(jsFile, context);
+        attUrl = getAttUrlCmd(extJsFileParams.getAttUrlCmd()).getAttachmentURL(
+            extJsFileParams.getJsFile(),
+            getModelContext().getXWikiContext());
       }
-      if (!StringUtils.isEmpty(params)) {
+      if (!StringUtils.isEmpty(extJsFileParams.getParams())) {
         if (attUrl.indexOf("?") > -1) {
-          attUrl += "&" + params;
+          attUrl += "&" + extJsFileParams.getParams();
         } else {
-          attUrl += "?" + params;
+          attUrl += "?" + extJsFileParams.getParams();
         }
       }
-      return addExtJSfileOnce_internal(jsFile, attUrl);
+      return Strings.nullToEmpty(addExtJSfileOnceInternal(extJsFileParams.getJsFile(),
+          attUrl, extJsFileParams.getLoadMode()));
     }
     return "";
   }
 
-  private String addExtJSfileOnce_internal(String jsFile, String jsFileUrl) {
+  private String addExtJSfileOnceInternal(String jsFile, String jsFileUrl, JsLoadMode loadMode) {
```

Comment:
the "internal" isn't needed (anymore)
```suggestion
  private String addExtJSfileOnce(String jsFile, String jsFileUrl, JsLoadMode loadMode) {
```